clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -61,32 +63,23 @@ public class MembershipManagerImplTest {
     private static final String MEMBER_ID = "test-member-1";
     private static final int MEMBER_EPOCH = 1;
 
+    private final LogContext logContext = new LogContext();
     private SubscriptionState subscriptionState;
     private ConsumerMetadata metadata;
-
     private CommitRequestManager commitRequestManager;
 
-    private ConsumerTestBuilder testBuilder;
-
     @BeforeEach
     public void setup() {
-        testBuilder = new ConsumerTestBuilder(ConsumerTestBuilder.createDefaultGroupInformation());
-        metadata = testBuilder.metadata;
-        subscriptionState = testBuilder.subscriptions;
-        commitRequestManager = testBuilder.commitRequestManager.get();
-    }
-
-    @AfterEach
-    public void tearDown() {
-        if (testBuilder != null) {
-            testBuilder.close();
-        }
+        // The SubscriptionState cannot be mocked, but needs to stay as a spy as the test relies on its logic.
+        subscriptionState = spy(new SubscriptionState(logContext, OffsetResetStrategy.EARLIEST));
```

Comment:
Subscription state is mostly a data container. I wonder if we could just insert and retrieve the expected data instead of injecting it via the spy?